rewrite-java/src/main/java/org/openrewrite/java/tree/Flag.java

```diff
@@ -33,6 +33,10 @@ public enum Flag {
     Interface(1L << 9),
     Abstract(1L << 10),
     Strictfp(1L << 11),
+    /**
+     * An enumeration type (on a class) or an enumeration constant (on a member)
+     */
+    Enum(1L << 14),
```

Comment:
Should also be added to the `VALID_CLASS_FLAGS` field below

Reply:
Right, that'd be more consistent, although the Class' Kind is set using the flag so that information is already available for a Class: https://github.com/openrewrite/rewrite/blob/45c438972d37db6b1f1903c87c752e141e4cc997/rewrite-java/src/main/java/org/openrewrite/java/internal/JavaReflectionTypeMapping.java#L123-L139

Reply:
Does the java compiler internally place that flag on enum classes? If so we should list it in our valid flags map. Otherwise leave it omitted.

Reply:
Approved and ok to merge if indeed we add the `Enum` flag to `VALID_CLASS_FLAGS`, and update the test to assert it's there, just to guard against regression.